x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/transport/actions/TransportGetWatcherSettingsAction.java

```diff
@@ -37,6 +38,7 @@ public class TransportGetWatcherSettingsAction extends TransportLocalClusterStat
     GetWatcherSettingsAction.Response> {
 
     private final IndexNameExpressionResolver indexNameExpressionResolver;
+    private final ProjectResolver projectResolver;
```

Comment:
You can make this class extend `TransportLocalProjectMetadataAction` instead of storing the field here.